src/solvers/dgmulti/flux_differencing_gauss_sbp.jl

```diff
@@ -25,15 +25,262 @@ function tensor_product_quadrature(element_type::Hex, r1D, w1D)
+end
 
+@inline function tensor_product_gauss_face_operator!(out::AbstractMatrix,
+                                                     A::TensorProductGaussFaceOperator,
+                                                     x::AbstractMatrix)
+  @threaded for col in Base.OneTo(size(out, 2))
+    tensor_product_gauss_face_operator!(view(out, :, col), A, view(x, :, col))
+  end
+end
 
 # By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
 # Since these FMAs can increase the performance of many numerical algorithms,
 # we need to opt-in explicitly.
 # See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
 @muladd begin
 
+# Interpolates values from volume Gauss nodes to face nodes on one element.
+@inline function tensor_product_gauss_face_operator!(out::AbstractVector,
+                                                     A::TensorProductGaussFaceOperator{2, Interpolation},
+                                                     x::AbstractVector)
+
+  @unpack interp_matrix_gauss_to_face_1d, face_indices_tensor_product = A
+  @unpack nnodes_1d, nfaces = A
+
+  fill!(out, zero(eltype(out)))
+  x = reshape(x, nnodes_1d, nnodes_1d)
+
+  # interpolation in the x-direction
+  @turbo for i in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, 1]
+    index_right = face_indices_tensor_product[2, i, 1]
+    for jj in Base.OneTo(nnodes_1d)      # loop over "line" of volume nodes
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[jj, i]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[jj, i]
+    end
+  end
+
+  # interpolation in the y-direction
+  @turbo for i in Base.OneTo(nnodes_1d) # loop over nodes in a face
+    index_left = face_indices_tensor_product[1, i, 2]
+    index_right = face_indices_tensor_product[2, i, 2]
+    for jj in Base.OneTo(nnodes_1d)               # loop over "line" of volume nodes
+      out[index_left] = out[index_left] + interp_matrix_gauss_to_face_1d[1, jj] * x[i, jj]
+      out[index_right] = out[index_right] + interp_matrix_gauss_to_face_1d[2, jj] * x[i, jj]
+    end
+  end
+end
+
+# Interpolates values from volume Gauss nodes to face nodes on one element.
+@inline function tensor_product_gauss_face_operator!(out::AbstractVector,
+                                                     A::TensorProductGaussFaceOperator{3, Interpolation},
+                                                     x::AbstractVector)
+
+  @unpack interp_matrix_gauss_to_face_1d, face_indices_tensor_product = A
+  @unpack nnodes_1d, nfaces = A
+
+  fill!(out, zero(eltype(out)))
+
+  # for 3D GaussSBP nodes, the indexing is first in y, then x, then z.
+  x = reshape(x, nnodes_1d, nnodes_1d, nnodes_1d)
+
+  # interpolation in the y-direction
+  @turbo for i in Base.OneTo(nnodes_1d), j in Base.OneTo(nnodes_1d) # loop over nodes in a face
```

Comment:
```suggestion
  @turbo for j in Base.OneTo(nnodes_1d), i in Base.OneTo(nnodes_1d) # loop over nodes in a face
```
`@turbo` will probably decide to re-arrange the loops this way but I would prefer keeping the consistent ordering in Trixi.jl.